File: Data Structure/Stack/LET_no402_Remove K Digits.py
class Solution:
    def removeKdigits(self, num: str, k: int) -> str:
        if k >= len(num):
            return "0"
        
        stack = []
        for c in num:
            while k and stack and stack[-1] > c:
                stack.pop()
                k -= 1
            stack.append(c)
        while k:
            stack.pop()
            k -= 1
        
        i = 0
        while i < len(stack) and stack[i] == '0':
            i += 1
        
        return ''.join(stack[i:]) if i < len(stack) else "0"

File: Data Structure/Stack/Medium/LET_no402_Remove K Digits.cpp
class Solution {
public:
    string removeKdigits(string num, int k) {
        int n = num.size();
        if (k >= n)
            return "0";

        string res;
        for (char c : num) {
            while (k > 0 && !res.empty() && res.back() > c) {
                res.pop_back();
                k--;
            }
            res.push_back(c);
        }

        while (k > 0) {
            res.pop_back();
            k--;
        }

        int i = 0;
        while (i < res.size() && res[i] == '0')
            i++;

        return i < res.size() ? res.substr(i) : "0";
    }
};